doc/tools/docgram/doc_grammar.ml

```diff
@@ -909,11 +909,12 @@ let apply_splice g edit_map =
   List.iter (fun b ->
       let (nt0, prods0) = b in
       let rec splice_loop nt prods cnt =
-        let max_cnt = 10 in
         let (nt', prods') = edit_rule g edit_map nt prods in
-        if cnt > max_cnt then
-          error "Splice for '%s' not done after %d iterations\n" nt0 max_cnt;
-        if nt' = nt && prods' = prods then
+        if cnt >= 10 then begin
+          error "Splice for '%s' not done after %d iterations.  Current value is:\n" nt0 cnt;
+          List.iter (fun prod -> Printf.eprintf "  %s\n" (prod_to_str prod)) prods';
+          (nt', prods')
+        end else if nt' = nt && prods' = prods then
           (nt', prods')
         else
           splice_loop nt' prods' (cnt+1)
```

Comment:
It seems that the result of calling `edit_rule` has nothing to do with the decision to stop iterating. Therefore, it would make more sense to reorder the call as in:

```ocaml
      let rec splice_loop nt prods cnt =
        if cnt >= 10 then
        begin
          error "Splice for '%s' not done after %d iterations.  Current value is:\n" nt0 cnt;
          List.iter (fun prod -> Printf.eprintf "  %s\n" (prod_to_str prod)) prods';
          (nt', prods')
        end
        else
          let (nt', prods') = edit_rule g edit_map nt prods in
          if nt' = nt && prods' = prods then (nt', prods')
          else splice_loop nt' prods' (cnt+1)
```

Reply:
Gotcha.  Updated.